mllam_data_prep/create_dataset.py

```diff
@@ -230,6 +239,26 @@ def create_dataset(config: Config):
         )
         ds["splits"] = da_splits
 
+    # ensure any dimensions for which coordinate values aren't yet set that
+    # these are given integer values. This will for example apply when stacking
+    # (x, y)-coordinates to a grid-index coordinate. These need unique values
+    # for later reference.
+    for d in ds.dims:
+        if d not in ds.coords:
+            ds[d] = np.arange(ds[d].size)
+
+    if config.output.domain_cropping is not None:
+        domain_cropping = config.output.domain_cropping
+        ds_interior_domain = create_dataset(config=config_interior_domain)
+        logger.info(
+            f"Cropping dataset to within convex hull margin of {ds_interior_domain} with a margin of {domain_cropping.margin_width_degrees} degrees"
+        )
+        ds = crop_to_within_convex_hull_margin(
+            ds=ds,
+            ds_reference=ds_interior_domain,
+            max_dist=domain_cropping.margin_width_degrees,
+        )
```

Comment:
At the moment this seems to introduce a `grid_index` coordinate for the `ds["splits"]` variable, that should not be there. This also causes errors downstream in neural-lam.

Reply:
Digging a bit more here I am not sure if it actually is the cropping that causes this. It still is an issue, but I can't figure out what causes it.

Reply:
I will look into it